tutorials/composite_bo_with_hogp.ipynb

```diff
@@ -0,0 +1,646 @@
+    "\n",
+    "Recently, [Maddox et al, '21](https://arxiv.org/abs/2106.12997) proposed a method for computing posterior samples from the HOGP by exploiting structure in the posterior distribution, thereby enabling its usage in BO settings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "I0712 081346.940 font_manager.py:1349] generated new fontManager\n"
+     ]
+    }
+   ],
+   "source": [
+    "import torch\n",
+    "import logging\n",
+    "import numpy as np\n",
+    "import math\n",
+    "import matplotlib.pyplot as plt\n",
+    "import time\n",
+    "\n",
+    "import gpytorch.settings as gpt_settings\n",
+    "\n",
+    "from botorch.acquisition import qSimpleRegret, qExpectedImprovement\n",
+    "from botorch.acquisition.objective import GenericMCObjective\n",
+    "from botorch.models import HigherOrderGP, SingleTaskGP\n",
+    "from botorch.models.higher_order_gp import FlattenedStandardize\n",
+    "from botorch.models.transforms import Normalize, Standardize\n",
+    "from botorch import fit_gpytorch_model\n",
+    "from botorch.optim import optimize_acqf\n",
+    "from botorch.sampling import IIDNormalSampler\n",
+    "from botorch.utils.sampling import draw_sobol_samples\n",
+    "\n",
+    "from gpytorch.mlls import ExactMarginalLogLikelihood\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Set Device and dtype"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Using  cpu\n"
+     ]
+    }
+   ],
+   "source": [
+    "device = torch.device(\"cpu\") if torch.cuda.is_available() else torch.device(\"cuda:0\")\n",
+    "dtype = torch.float\n",
+    "\n",
+    "print(\"Using \", device)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "models_used = (\n",
+    "    \"rnd\",\n",
+    "    \"ei\",\n",
+    "    \"ei_hogp_cf\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Problem Description"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We use a simple test problem describing the concentration of pollutants after a chemical spill from [Astudillo & Frazier, ICML, '19](https://proceedings.mlr.press/v97/astudillo19a.html) defined over a $3 \\times 4$ grid of values $s,t$ and we wish to optimize the parameters w.r.t. their true values, to estimate the true value of parameters, $x = [M, D, L, \\tau].$ The function is given by \n",
```

Comment:
maybe note here that it's important for our setup that the function is diff'able and implemented in torch so we can backprop through it.